src/libs/actions/IOU.ts

```diff
@@ -3680,6 +3664,7 @@ function requestMoney(
         }
     }
 
+    InteractionManager.runAfterInteractions(() => TransactionEdit.removeDraftTransaction(CONST.IOU.OPTIMISTIC_TRANSACTION_ID));
```

Comment:
Let's not add that. Please remove it (same for other cases)

Reply:
That was my initial thought but there is an extremely edge case of the user accessing via direct link on which case the flow can start on previous uncleared transaction draft.
And also two other cases which we directly navigate to expense flow, on which case we can ofcourse clear draft transaction before navigating
1. https://github.com/Expensify/App/blob/58258ed4d00588888c00296778bd919dc6d3a4a1/src/pages/iou/request/IOURequestRedirectToStartPage.tsx#L30-L37
2.
https://github.com/Expensify/App/blob/58258ed4d00588888c00296778bd919dc6d3a4a1/src/libs/Navigation/Navigation.ts#L108-L112
Only the direct link edge case is the unavoidable one
 WDYT

Reply:
I don't think this is the best approach. It may be better to just fix the timing. Can you try change MERGE to SET (or viceversa) for the clearing instruction

Reply:
So that the navigation happen before the onyx value is cleared

Reply:
That is what I have done @s77rt as the clearing is happening after interaction.

Reply:
`InteractionManager.runAfterInteractions` does not have a clear timing on when it will be called and it should be avoided in general in web. Can you try remove it and just change the onyx clearing instruction to use the MERGE method instead of SET?

Reply:
You are saying MERGE is slower? The problem is BTW I couldn't reproduce it even on main because my emulator is faster I think. You know this kinds of issues occur when the animation is too slow but in my case it is not slow enough therefore I cannot determine if MERGE will fix it. Moreover, even if `MERGE` might be slow enough to fix for you it might still be reproducible on a slower device.
I think I am following the right approach and we solved similar problem with this approach and used runAfterInteraction in all platforms. Here is an example:
https://github.com/Expensify/App/blob/58258ed4d00588888c00296778bd919dc6d3a4a1/src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx#L64-L66

Reply:
> You are saying MERGE is slower?

Yes (more like delayed), AFAIK merge operations are batched and run async where set operations are sync.

> I couldn't reproduce it even on main

Can you please check again in different browsers or try on iOS (Safari)

Reply:
Still reproduced it with MERGE on IOS safari

https://github.com/user-attachments/assets/f3489ede-f018-4bfb-8af4-676adc4c63b6

Reply:
Can we avoid the onyx data separation? Functions such as `splitBill` are not aware of the transaction and it's not clear that calling `removeDraftTransaction` is required. This also lead to bugs as now we have more functions where we are not clearing the transaction e.g. `shareTrackedExpense`

Reply:
I am not fully understanding what you want but to clarify things: I tracked back every expense creating functions that use optimistic data returned from buildingOnyxData kind of functions that were resetting the draft transaction.
For instance, for split bill we are using an optimistic data returned here
https://github.com/Expensify/App/blob/512ae9f1a1c6d840065f377264bc1ab009356f0f/src/libs/actions/IOU.ts#L4344
which used to clear the draft transaction here
https://github.com/Expensify/App/blob/512ae9f1a1c6d840065f377264bc1ab009356f0f/src/libs/actions/IOU.ts#L627-L632

Reply:
`shareTrackedExpense` and other functions used to clear the draft but now they are not

Reply:
We are only concerned with draft transaction with key `CONST.IOU.OPTIMISTIC_TRANSACTION_ID` and I have tried to include all cases and shareTrackedExpense wasn't clearing it on main too 👍

Reply:
It does clear it on main. `shareTrackedExpense` `onyxData` is coming from `getTrackExpenseInformation` <- `buildOnyxDataForTrackExpense`

Reply:
But that is not inside shareTrackedExpense that's why you confused me but we are already clearing here
https://github.com/Expensify/App/blob/cc5d41b827444cc1695762ca5328385fc6c81ad3/src/libs/actions/IOU.ts#L3920

Reply:
Ahi right my bad!